geoh5py/ui_json/enforcers.py

```diff
@@ -314,24 +315,16 @@ def enforce(self, name: str, value: Any):
         if not self.rule(value):
             raise self.validation_error(
                 name,
-                [k for k in self.validations if k not in value],
+                [k for k in self.validations if k not in self.collection(value)],
             )
 
     def rule(self, value: Any) -> bool:
         """True if all required parameters are in 'value' collection."""
-        return all(k in value for k in self.validations)
-
-    @property
-    def validations(self) -> list[str]:
-        if self._validations is None:
-            self._validations = []
-        return self._validations
+        return all(k in self.collection(value) for k in self.validations)
 
-    @validations.setter
-    def validations(self, val: str | list[str]):
-        if isinstance(val, str):
-            val = [val]
-        self._validations = val
+    def collection(self, value: Any) -> list[Any]:
+        """Returns the collection to check for required items."""
+        return value

```

Comment:
What's happening here, the `collection` doesn't seem to do anything?

Reply:
Just in the base class.  I use this in the children to transform the collection.  Eg RequiredWorkspaceObjectEnforcer takes the dictionary of param/value from uijson as it's value, but the collection that I'm interested in is the geoh5.list_entities_name.